core/src/epicli/data/common/ansible/playbooks/roles/repository/files/download-requirements/common/common_functions.sh

```diff
@@ -0,0 +1,122 @@
+echol() {
+#
+# Print to stdout, optionally to a log file.
+# Requires $CREATE_LOGFILE and $LOG_FILE_PATH to be defined.
+#
+# param $@: args which will be printed
+#
+    echo -e "$@"
+    if [[ $CREATE_LOGFILE == "yes" ]]; then
+        local timestamp=$(date +"%b %e %H:%M:%S")
+        echo -e "${timestamp}: $@" >> "$LOG_FILE_PATH"
+    fi
+}
+
+
+exit_with_error() {
+    echol $@
+    exit 1
+}
+
+
+__check_curl() {
+#
+# Use curl in silent mode to check if target `url` is available.
+#
+# param $1: url which will be tested
+#
+    [[ $# -lt 1 ]] && exit_with_error "(curl), no url provided"
+    local url=$1
+
+    echol "Testing connection to: ($url) using: (curl)..."
+
+    err_msg=$(curl -S --silent $url 2>&1 >/dev/null) || exit_with_error "(curl) failed, reason: ($err_msg)"
```

Comment:
```suggestion
    err_msg=$(curl --show-error --silent $url 2>&1 >/dev/null) || exit_with_error "curl failed, reason: ($err_msg)"
```
IMO in scripts it's better to use options in long form unless they are commonly known (intuitive/popular).